src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/CompletionItemExtensions.cs

```diff
@@ -60,15 +65,33 @@ public static VSCompletionItem ToVSCompletionItem(this CompletionItem completion
                 Detail = completion.Detail,
                 Documentation = completion.Documentation,
                 FilterText = completion.FilterText,
-                InsertText = completion.FilterText,
+                InsertText = completion.InsertText,
                 InsertTextFormat = completion.InsertTextFormat,
                 Kind = completion.Kind,
                 Label = completion.Label,
                 Preselect = completion.Preselect,
                 SortText = completion.SortText,
                 Tags = completion.Tags,
                 TextEdit = completion.TextEdit,
+                VsCommitCharacters = vsCommitCharacters,
             };
+
+            static IEnumerable<VSCommitCharacter> GetVSCommitCharacters(CompletionItem completion)
+            {
+                if (completion.CommitCharacters is null)
+                {
+                    yield break;
```

Comment:
`yield` returning / breaking has benefits when there's a chance that you don't iterate through all items. If you do often iterate all items then you're better off building the return value / array yourself from a perf perspective.